sdk/storage/Azure.Storage.Files.DataLake/tests/DataLakeTestBase.cs

```diff
@@ -379,6 +380,37 @@ public async Task<string> SetupFileSystemLeaseCondition(DataLakeFileSystemClient
             return leaseId == ReceivedLeaseId ? lease.LeaseId : leaseId;
         }
 
+        /// <summary>
+        /// Gets a custom account SAS where the permissions, services and resourceType
+        /// comes back in the string character order that the user inputs it as.
+        /// </summary>
+        /// <param name="permissions"></param>
+        /// <param name="services"></param>
+        /// <param name="resourceType"></param>
+        /// <param name="sharedKeyCredential"></param>
+        /// <returns></returns>
+        public string GetCustomAccountSas(
```

Comment:
I threw this method in every service test base to prevent having to include all the necessary shared SAS test resources, that took a dependency on some SAS storage shared resources. Let me try again to throw this method into shared test code and remember why it was painful to do, or if I was just having a bad day with dependencies.

Also FWIW So for DataLake specifically, it doesn't allow you to set the immutable policy ('i) in the permissions. So if you attempted to send it with an 'i' in the permissions the service throws an error. So the default permissions does not include the 'i' in comparison to the rest of the services. I mean what we can do is just not include 'i' in the permissions for all of them because I doubt any of the tests require the 'i' by default.

Reply:
Okay so this still kinda exists in DataLakeTestBase BUT because we need to take in the HNS credentials instead of the Primary credentials